code/_core/obj/structure/interactive/diverter.dm

```diff
@@ -31,11 +43,19 @@
 			else
 				caller.to_chat(span("notice","You anchor the diverter."))
 				anchored = TRUE
+		else if(T.flags_tool & FLAG_TOOL_MULTITOOL && mode == MODE_VALUE) // Should I REALLY make it need a multitool?
+			value_threshold = clamp(input(caller, "What do you want to set the threshold to?","[src]",value_threshold),-1,50000)
+			caller.to_chat(span("notice","You set the threshold to [value_threshold]."))
 	else
-		caller.visible_message(span("notice","\The [caller.name] rotates \the [src.name]."),span("notice","You rotate \the [src.name]."))
-		set_dir(turn(dir,90))
+		if(caller.attack_flags & CONTROL_MOD_DISARM)
+			var/modes = list(MODE_NONE,MODE_VALUE,MODE_DENSITY,MODE_MATERIAL,MODE_LIVING,MODE_WEAPON,MODE_CRATE,MODE_BUTCHERABLE)
+			mode = input(caller, "What do you want to set the mode to?","[src]",mode) as null|anything in modes
+			name = "[initial(name)] ([mode])"
```

Comment:
Should have an option in the modes list that has "Cancel" in the end, as well as a check to make it so that null mode inputs will set the mode to MODE_NONE as clicking cancel in the input pop-up will cause errors.